lib/pbench/server/api/resources/query_apis/datasets_detail.py

```diff
@@ -158,12 +157,11 @@ def postprocess(self, es_json: JSON, context: CONTEXT) -> JSON:
         try:
             m = self._get_metadata(src["run"]["name"], context["metadata"])
         except DatasetNotFound:
-            abort(
-                HTTPStatus.BAD_REQUEST,
-                message=f"Dataset {src['run']['name']} not found",
+            raise APIAbort(
+                HTTPStatus.BAD_REQUEST, f"Dataset {src['run']['name']} not found"
```

Comment:
The `_get_metadata` method takes the dataset name and tries to look it up, so `DatasetNotFound` means the *dataset* isn't there. That's "impossible" at this point since we're in postprocess and we already found it in preprocess; but if it happens, this is now an internal error because something beyond the user's control happened. (Although, technically, maybe it's HTTPStatus.GONE (410), because it was there but someone else poached it with a DELETE in the middle of your query.)

A metadata access error would be `MetadataError` (handled below).

And, since this code was written, I ended up splitting `_get_metadata` as a wrapper (which looks up the dataset) around `_get_dataset_metadata` which takes the `Dataset` object instead of looking it up again. I believe we have that in the `context` object here, so we could really be using that ... not a big deal for this PR, however.